R/apa_print_manova.R

```diff
@@ -0,0 +1,115 @@
+
+#' @param test For MANOVA, the multivariate test statistic to be reported, see \code{\link{summary.manova}}.
+#' @rdname apa_print.aov
+#' @export
+
+apa_print.manova <- function(x, test = "Pillai", ...) {
+
+  tidied_x <- as.data.frame(
+    x = broom::tidy(x, test = test, ...)
+    , stringsAsFactors = FALSE
+  )
+  print_manova(x = tidied_x, ...)
+}
+
+
+#' @rdname apa_print.aov
+#' @export
+
+apa_print.summary.manova <- function(x, ...) {
+
+  tidied_x <- as.data.frame(
+    x = broom::tidy(x, ...)
+    , stringsAsFactors = FALSE
+  )
+  print_manova(x = tidied_x, ...)
+}
+
+
+
+#' @keywords internal
+
+print_manova <- function(
+  x
+  , in_paren = FALSE
+) {
+
+
+
+  estimate_names <- c(
+    pillai = "V"
+    , wilks = "\\Lambda"
+    , hl = "T"
+    , roy = "\\theta"
+  )
+
+
+  estimate_names_with_dollars <- paste0("$", estimate_names, "$")
+  names(estimate_names_with_dollars) <- names(estimate_names)
+
+  multivariate_stat <- intersect(names(estimate_names), colnames(x))
+
+  if(length(multivariate_stat) == 0L) {
+    stop("No multivariate test statistic found in results object.")
+  } # move this to input validation
+
+  col_names <- c(
+    term = "Effect"
+    , df = "df"
+    , pillai = "pillai"
+    , wilks = "wilks"
+    , hl = "hl"
+    , roy = "roy"
+    , statistic = "F"
+    , num.df ="df1"
+    , den.df = "df2"
+    , p.value ="p"
+  )
+  colnames(x) <- col_names[colnames(x)]
+
+  var_labels <- c(
+    estimate_names_with_dollars
+    , "Effect" = "Effect"
+    , "F" = "$F$"
+    , "df1" = "$\\mathit{df}_1$"
+    , "df2" = "$\\mathit{df}_2$"
+    , "p" = "$p$"
+  )
+
+  x <- x[x$Effect != "Residuals", colnames(x) != "df", drop = FALSE]
+
+  x[["Effect"]] <- prettify_terms(x[["Effect"]])
+  x[[multivariate_stat]] <- printnum(x[[multivariate_stat]])
+  x[["F"]] <- printnum(x[["F"]])
+  x[["df1"]] <-as.character(round(x[["df1"]], digits = 2))
```

Comment:
Out of curiosity: Why don't you use `printnum()` here?

Reply:
I'm not aware of any correction methods for degrees of freedom in MANOVA, but this way the default is the same as for ANOVA: If integer df are provided, no decimals are reported; if there are corrected ones, they will be reported rounded up to two digits (edge cases where only one digit is significant, but two should be printed are not captured, yet).

Reply:
Ah, I see. The downside of this approach is that any df that come out as having exactly 1 digit will be printed with only one digit. How about something like this:

~~~r
df <- c(1, 2, 1.1, 1.35987, 5.210938)
printnum(df, digits = as.numeric(df %% 1 != 0) * 2)
~~~